### ISSUE TITLE
NVDA installer addons can produce an updater loop

### ISSUE THREAD

**Issue by christopherpross** [nvaccess/nvda#16903]
### Steps to reproduce:
1. Download the sample GlobalPlugin that I am attached to this issue. This plugin just display a dialog with yes and no at every start of NVDA.
2. Place this plugin in your scratchpath and trigger an update of NVDA, for example from 2024.3beta3 to 2024.3beta4.
3. confirm the UAC dialog with yes and wait that the please wait dialog disapears.

### Actual behavior:

The update fails, because the question dialog is open and nvda can not restart or exited.

### Expected behavior:

In case of an update, NVDA should able to restart, maybe we can give the user the opertunity to close the dialog before we continue with the update or we exit NVDA anyways and ignoring all other NVDA dialogs.

### NVDA logs, crash dumps and other attachments:
`
INFO - core.main (16:43:14.044) - MainThread (30692):
NVDA initialized
INFO - brailleDisplayDrivers.handyTech.BrailleDisplayDriver.__init__ (16:43:15.692) - ThreadPoolExecutor-0_0 (35468):
Found Modular Evolution 88 connected via serial (COM3)
INFO - braille.BrailleHandler._setDisplay (16:43:15.694) - ThreadPoolExecutor-0_0 (35468):
Loaded braille display driver 'handyTech', current display has 88 cells.
WARNING - RPC process 11228 (nvda_slave.exe) (16:43:21.360) - Dummy-14 (25760):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed after a reboot or the  installer will try removing it again next time. Directory: 'C:\\Program Files (x86)\\NVDA\\lib\\2024.3beta4'
WARNING - RPC process 11228 (nvda_slave.exe) (16:43:27.401) - Dummy-14 (25760):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed after a reboot or the  installer will try removing it again next time. Directory: 'C:\\Program Files (x86)\\NVDA\\lib64\\2024.3beta3'
WARNING - RPC process 11228 (nvda_slave.exe) (16:43:27.407) - Dummy-14 (25760):
installer.removeOldLibFiles:
Failed to remove a directory no longer needed. This can be manually removed after a reboot or the  installer will try removing it again next time. Directory: 'C:\\Program Files (x86)\\NVDA\\lib64\\2024.3beta4'
ERROR - core.triggerNVDAExit (16:43:28.356) - MainThread (30692):
NVDA cannot exit safely, ensure open dialogs are closed
ERROR - gui.installerGui.doInstall (16:43:28.508) - MainThread (30692):
NVDA already in process of exiting, this indicates a logic error.
`

#### NVDA installed/portable/running from source:
NVDA installed, 
#### NVDA version:
reproduced with 2024.3 beta 3, beta 4, alpha-32997,3f82ab67 
#### Windows version:
Version 23H2 (Build 22631.3880)
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes, the only way to exit of the updater loop is to manualy remove the addon from the users directory.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, I tested the following versions, but the behavior was the same:
1. 2024.3 beta 3
2. 2024.3 beta 4
3. alpha-32997,3f82ab67 

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes, that is strange, if I choose in the exit dialog the option to restart with deactivated addons after the first error, this setting is not applied. The addon is also active and spawns the dialog.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by christopherpross** [comment]
Here the sample addon, I don't saw the message from GitHub that .py files are not allowed. Simple remove the '.txt' extension.
[test.py.txt](https://github.com/user-attachments/files/16363995/test.py.txt)

**Comment by XLTechie** [comment]
@christopherpross Is the behavior different when following your STR with a portable copy, than it is when following it in an installed copy?

I updated a portable copy with your test scratchpad code installed.
The temporary copy failed to exit, because it can't while a `gui.message.messageBox` is open.
However, after I closed the messageBox, and manually exited, the portable copy had been updated.

If I instead had restarted, it is true that the installer temporary copy would have restarted.

But in neither case would the update have failed, it just wouldn't start running after the update, which I agree is a problem.
However your "Actual behavior" section, says that the update failed. Are you sure it did?

**Comment by christopherpross** [comment]
@XLTechie  
Sorry for the delay. 
If I try the Steps to reproduce with a portable copy with NVDA, I get exactly the behaviour that you described.
If I exit manualy, it will start normaly, if I use restart, the installer updates the portable copy a second time.
The interesting part here, when I use an installed copy, it doesn't matter which option I coose in the exit dialog, everytime the installer is started and installs the new version.

You are right, I checked now the version, I used my test addon to print out the version and the nvda, in both scenarios is already updated, only the installer is simple executed.

So the main issue is, that in case of the installed version all options and in case of the portable version, only restart brings the installer a second time up. That is the problem, NVDA is already updated.
